src/include/sof/audio/component.h

```diff
@@ -223,70 +222,159 @@ enum comp_copy_type {
 struct comp_driver;
 
 /**
- * Audio component operations - all mandatory.
+ * Audio component operations.
  *
  * All component operations must return 0 for success, negative values for
- * errors and 1 to stop the pipeline walk operation.
+ * errors and 1 to stop the pipeline walk operation unless specified otherwise
+ * in the operation documentation.
  */
 struct comp_ops {
-	/** component creation */
+	/**
+	 * Creates a new component device.
+	 * @param drv Parent component driver.
+	 * @param comp Component parameters.
+	 * @return Pointer to the new component device.
+	 *
+	 * All required data objects should be allocated from the run-time
+	 * heap (RZONE_RUNTIME).
+	 * Any component-specific private data is allocated separately and
+	 * pointer is connected to the comp_dev::private by using
+	 * comp_set_drvdata() and later retrieved by comp_get_drvdata().
+	 *
+	 * All parameters should be initialized to their default values.
+	 */
```

Comment:
I would suggest on first factoring out everything that needs to go to component_ext and then enhance documentation. Otherwise, it is hard to understand the intention here.

Reply:
Moved dox related changes to a separate patch.
cc @paulstelian97